TypeSystem/Schema/smoke_test_struct.h

```diff
@@ -20,6 +21,7 @@ CURRENT_STRUCT(Primitives) {
   CURRENT_FIELD(m, bool);
   CURRENT_FIELD(n, std::chrono::microseconds);
   CURRENT_FIELD(o, std::chrono::milliseconds);
+  CURRENT_FIELD_DESCRIPTION(b, "Field descriptions can be set in any order.");
```

Comment:
For F# and C, I'd suggest to place the description on the line right before the field, and separate such described fields from the other ones with empty lines. BTW, escaping of `'` is not needed in the comments.
```
type Primitives = {
  // It's the order of fields that matters.
  a : byte

  // Field descriptions can be set in any order.
  b : uint16

  c : uint32
  d : uint64
  e : sbyte
  f : int16
  g : int32
  h : int64
  i : char
  j : string
  k : float
  l : double

  // Multiline
  // descriptions
  // can be used.
  m : bool

  // microseconds.
  n : int64

  // milliseconds.
  o : int64
}
```